src/core/Akka/Pattern/FutureTimeoutSupport.cs

```diff
@@ -0,0 +1,40 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="BackoffSupervisor.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Akka.Actor;
+
+namespace Akka.Pattern
+{
+    public static class FutureTimeoutSupport
+    {
+        /// <summary>
+        /// Returns a Task that will be completed with the success or failure of the provided value after the specified duration.
+        /// </summary>
+        /// <typeparam name="T">TBD</typeparam>
+        /// <param name="value">TBD</param>
+        /// <param name="timeout">TBD</param>
+        /// <param name="scheduler">TBD</param>
+        public static Task<T> After<T>(TimeSpan timeout, IScheduler scheduler, Func<Task<T>> value)
```

Comment:
I'm not sure if we need to move this as a pattern. If this would be a potential patter it should probably be an extension method with signature like `Task<T> After<T>(this IScheduler scheduler, TimeSpan timeout, Func<Task<T>> call)`. But so far it's going to be used only in one place. I'd keep it encapsulated within entity recovery strategy.